### PR TITLE
@damassi => Add flag to be able to timeout Redis get calls

### PR DESCRIPTION
**Pull request by mzikherman**
This adds a flag `CACHE_RETRIEVAL_TIMEOUT_MS` (defaults to 2000ms), which will timeout calls. The idea (all from https://medium.com/@stockholmux/tick-tock-setting-a-timeout-on-a-redis-call-with-node-js-be63733df98a), is to wrap that call and callback to our code which is waiting for it with an error (which is then treated as a cache miss). Otherwise we clear that interval and return the data normally.

Running locally w/ aggressive timeouts (<10ms), and immediate timeouts (0ms), does show the behavior I'd expect (treated as a miss and making the request).

### CODE REVIEW

**Comment by mzikherman** [comment]
I couldn't find anything in the node Redis client that supported this, thus the blog post + hack.

File: src/lib/cache.js [hunk 1]
```diff
@@ -68,12 +69,35 @@ function createRedisClient() {
 
 export const client = isTest ? createMockClient() : createRedisClient()
 
+// Used to timeout Redis calls
+// From: https://medium.com/@stockholmux/tick-tock-setting-a-timeout-on-a-redis-call-with-node-js-be63733df98a
+function timeoutRedisCommand(command, timeoutValue) {
+  return function wrappedCommand() {
```
Comment (damassi) [inline src/lib/cache.js]: `wrappedCommand(...args)`

File: src/lib/cache.js [hunk 2]
```diff
@@ -68,12 +69,35 @@ function createRedisClient() {
...
+    const argsAsArray = Array.prototype.slice.call(arguments)
```
Comment (damassi) [inline src/lib/cache.js]: Can just be `wrappedCommand(...args)`

**Comment by damassi** [comment]
One little thing but not a blocker 👍

File: src/lib/cache.js [hunk 3]
```diff
@@ -68,12 +69,35 @@ function createRedisClient() {
...
+    argsAsArray.push((err, data) => {
+      clearTimeout(timeoutHandler)
+      callback(err, data)
+    })
+
+    client[command].apply(client, argsAsArray)
+  }
+}
+
+const getWithTimeout = timeoutRedisCommand("get", CACHE_RETRIEVAL_TIMEOUT_MS)
```
Comment (cavvia) [inline src/lib/cache.js]: :+1: nice pattern!

**Pull request merged**